assets/javascripts/discourse/components/chat-channel-row.js

```diff
@@ -59,6 +68,13 @@ export default Component.extend({
       return;
     }
 
+    if (
```

Comment:
could you explain the reason of this please ? I didn't dig but that seems hacky

Reply:
Full context:

```javascript
@action
handleClick(event) {
  if (event.target.classList.contains("chat-channel-leave-btn")) {
    return;
  }

  if (
    event.target.classList.contains("chat-channel-settings-btn") ||
    event.target.parentElement.classList.contains("select-kit-header-wrapper")
  ) {
    return;
  }

  this.handleSwitchChannel(event);
},
```

I am not a fan of this either, but without this special case I added the select-kit menu does not show when you click on this cog on hover:

![image](https://user-images.githubusercontent.com/920448/156947271-ff48da06-45b8-47b5-96c1-8f9efa5a7dc0.png)

Also, I assume the first condition is to make this X button work:

![image](https://user-images.githubusercontent.com/920448/156947299-4b3138b9-e9ba-4566-ab6e-28fee0a81e52.png)

If there is a way of making these both work without these weird class overrides I can change it, just wasn't sure how else to make the select-kit menu show up correctly on click in this situation.